bugzilla2gitlab/models.py

```diff
@@ -246,13 +252,13 @@ class Attachment(object):
     '''
     The attachment model
     '''
-    def __init__(self, bugzilla_attachment_id, filename):
+    def __init__(self, bugzilla_attachment_id, file_description):
         self.id = bugzilla_attachment_id
-        self.filename = filename
+        self.file_description = file_description
```

Comment:
It's a filename. You call it a filename in the `Comment` class, and it makes sense to be consistent.

Reply:
General comment: I realized that the change was not well structured and hard to follow so I've made a refactor. Please take a look. There are now 2 additional methods in Comment class and each method is kept short.

W.r.t. whether the original class member is a filename or description, I think it's a description. When you upload an attachment file, say `script.py` and describe it, say `Sample script`, Bugzilla will show `Sample script` in attachment table but when you click there your browser will download `script.py`.

My motivation was to preserve this way in gitlab. New code extracts file name (`script.py` in this example) from Content-disposition header same way as your browser does, then uses it when uploading the file to gitlab. Gitlab returns attachment markdown with file name used twice, as both the link text and the uploaded file url. We adjust it to use the original file description as the link text. This produces same markdown as in Bugzilla, and same file name is kept when downloading attachments. One side effect (in our deployment) is that localized file descriptions no longer break the migration, as long as file names are in ASCII. This might be typical for non-English speaking teams.